RUFAS/biophysical/animal/nutrients/beef_cow_calf_requirements_calculator.py

```
@@ -0,0 +1,516 @@
+            has days_pregnant set; if days_pregnant is out of 1–283 range; if
+            days_in_milk is negative; if a bull or non-female has days_in_milk > 0;
+            or if body_weight / mature_body_weight are not positive and finite.
+
+        """
+        if not inputs.animal_type.is_beef_cow_calf:
+            raise ValueError(
+                f"BeefCowCalfRequirementsCalculator only handles cow-calf types; " f"got {inputs.animal_type.value}."
+            )
+        if inputs.days_pregnant is not None:
+            if inputs.animal_type is AnimalType.BEEF_BULL:
+                raise ValueError("BEEF_BULL cannot have days_pregnant set; bulls do not gestate.")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)